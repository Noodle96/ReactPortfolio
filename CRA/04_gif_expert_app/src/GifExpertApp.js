import { Fragment, useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

//FUNCTIONAL COMPONENT
export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['messi']);
  
    return (
        <Fragment>
            <h2>GiftExpertApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr />
            {/* <button onClick={(e) => {handleAdd(e)}}>Add Category</button> */}
            <ol>
                {
                    categories.map( category => (
                         <GifGrid 
                            key={category}
                            category={category}
                         />
                    ))
                    
                }
            </ol>     
        </Fragment>
    );
};
