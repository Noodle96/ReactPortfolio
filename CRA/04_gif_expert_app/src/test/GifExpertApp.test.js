import {shallow} from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en el componente GifExpertApp', () => {
    test('Deberia hacer toMatchSnapshot correctamente', () => { 
        const wrapper = shallow(<GifExpertApp /> );
        expect(wrapper).toMatchSnapshot();
    });

    // test('DeberÃ­a de coincidir en elementos la lista de categorias', () => {
    //     const defaultCategories = ['xyz', 'abc'];
    //     const wrapper = shallow(<GifExpertApp defaultCategories={defaultCategories} /> );
    //     expect(wrapper).toMatchSnapshot();
    //     expect(wrapper.find('GifGrid').length).toBe(defaultCategories.length);
    // });
});