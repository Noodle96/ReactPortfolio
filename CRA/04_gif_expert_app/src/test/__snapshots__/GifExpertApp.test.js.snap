// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifExpertApp Deberia hacer toMatchSnapshot correctamente 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={
      Array [
        "messi",
      ]
    }
  />
  <hr />
  <ol />
</Fragment>
`;
